ts/visitor.ts

```diff
@@ -186,13 +186,15 @@ export const visitor = {
     },
     exit(path: NodePath<t.Function>, st: S) {
       st.elem.inConstructor = st.elem.inConstructorStack.pop()!
-      if ((path.node as any).shadow) {
+      if (path.has('shadow')) {
         //  Note(Sam L.) Babel arrow function transform leave shadow as true when
         // it transforms an arrow function to a function.
-        // The class transform relies on shadow to do extra work on arrow functions 
-        // (like using arguments that is in the outer scope instead of inner scope)
-        // That is unnecessary as it breaks aritychecking so it is set as undefined.
-        (path.node as any).shadow = undefined;
+        // Babel's shadow functions are just traditional functions but act like
+        // arrow functions (i.e has lexical binding of this and arguments)
+        // It's to let other plugins see if additional transform needs to be done
+        // on these functions. The classes transform would assume the arity
+        // checking we did was using lexical binding of arguments, but we're not.
+        (path.node as any).shadow = undefined; // therefore this is set to undefined
```

Comment:
I could not find any documentation on Babel's shadow functions so I am basing my info off of the comments in this issue: https://github.com/babel/babel/issues/2342
According to this PR, https://github.com/babel/babel/pull/5677, Babel has stopped using `.shadow` starting version 7. (If we do upgrade to Babel 7+, this check will most likely be not necessary)